src/hardware/pcspeaker_discrete.cpp

```diff
@@ -0,0 +1,491 @@
+					if (port_b_state.timerGateAndSpeakerOutput.all())
+						AddDelayEntry(delay_base, pit_last);
+					pit_index = 0;
+				} else {
+					pit_index += passed;
+					return;
+				}
+			} else {
+				if ((pit_index + passed) >= pit_half) {
+					const auto delay = pit_half - pit_index;
+					delay_base += delay;
+					passed -= delay;
+					pit_last = amp_positive;
+					if (port_b_state.timerGateAndSpeakerOutput.all())
+						AddDelayEntry(delay_base, pit_last);
+					pit_index = pit_half;
+				} else {
+					pit_index += passed;
+					return;
+				}
+			}
+		}
+		break;
+		// END CASE 2
+
+	case PitMode::SquareWaveAlias:
+	case PitMode::SquareWave:
+		while (passed > 0) {
+			/* Determine where in the wave we're located */
+			if (pit_index >= pit_half) {
+				if ((pit_index + passed) >= pit_max) {
+					const auto delay = pit_max - pit_index;
+					delay_base += delay;
+					passed -= delay;
+					pit_last = amp_positive;
+					if (port_b_state.timerGateAndSpeakerOutput.all())
+						AddDelayEntry(delay_base, pit_last);
+					pit_index = 0;
+					/* Load the new count */
+					pit_half = pit_new_half;
+					pit_max  = pit_new_max;
+				} else {
+					pit_index += passed;
+					return;
+				}
+			} else {
+				if ((pit_index + passed) >= pit_half) {
+					const auto delay = pit_half - pit_index;
+					delay_base += delay;
+					passed -= delay;
+					pit_last = amp_negative;
+					if (port_b_state.timerGateAndSpeakerOutput.all())
+						AddDelayEntry(delay_base, pit_last);
+					pit_index = pit_half;
+					/* Load the new count */
+					pit_half = pit_new_half;
+					pit_max  = pit_new_max;
+				} else {
+					pit_index += passed;
+					return;
+				}
+			}
+		}
+		break;
+		// END CASE 3
+
+	case PitMode::SoftwareStrobe:
+		if (pit_index < pit_max) {
+			/* Check if we're gonna pass the end this block */
+			if (pit_index + passed >= pit_max) {
+				const auto delay = pit_max - pit_index;
+				delay_base += delay;
+				// passed -= delay; // not currently used
+				pit_last = amp_negative;
+				if (port_b_state.timerGateAndSpeakerOutput.all())
+					AddDelayEntry(delay_base,
+					              pit_last); // No new events
+					                         // unless
+					                         // reprogrammed
+				pit_index = pit_max;
+			} else
+				pit_index += passed;
+		}
+		break;
+		// END CASE 4
+	case PitMode::HardwareStrobe:
+	case PitMode::Inactive:
+	default:
+		LOG_WARNING("PCSPEAKER: Unhandled PIT mode %s", PitModeToString(pit_mode));
+		break;
+	}
+}
+
+// PIT-mode activation
+void PcSpeakerDiscrete::SetCounter(int cntr, const PitMode mode)
+{
+	const auto newindex = PIC_TickIndex();
+	ForwardPIT(newindex);
+
+	prev_pit_mode = pit_mode;
+	pit_mode      = mode;
+	switch (pit_mode) {
+	case PitMode::InterruptOnTC: /* Mode 0 one shot, used with realsound */
+		if (!port_b_state.timerGateAndSpeakerOutput.all())
+			return;
+		if (cntr > 80) {
+			cntr = 80;
+		}
+		pit_last = ((double)cntr - 40) * (amp_positive / 40.0);
```

Comment:
Extract constant (40)

Reply:
Extracted, but don't know much more; added a comment to that effect.